### DISCUSSION TITLE
URL Dynamic

### DISCUSSION THREAD

**Discussion by DionatanLM** [socketio/socket.io#5129] [Q&A]
I need the URL to be variable but I don’t know how to do that.

```
export const socket = io('ws://backend', {
  autoConnect: false,
});
socket.io.opts.query = {
        user_id
 };
```
but I also wanted to be able to update the URL(Ws://backend) in the same way because I will not have a fixed URL, each user of my application can configure the route of the backend.

Anybody know a way I could do that?

**Comment by darrachequesne** [comment]
Hi! I don't think this is possible, as the URL is parsed internally.

You should rather create a new `socket` object every time the URL changes.
